src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -310,6 +313,23 @@ class Results extends React.Component<Props, State> {
     return url;
   };
 
+  handleIncompatibleQuery: React.ComponentProps<
+    typeof CreateAlertButton
+  >['onIncompatibleQuery'] = incompatibleAlertNoticeFn => {
+    const {organization} = this.props;
+    trackAnalyticsEvent({
+      eventKey: 'discover_v2.create_alert_incompatible',
+      eventName: 'Discoverv2: Creating an alert from discover was incompatible',
+      organization_id: parseInt(organization.id, 10),
+    });
+
+    const incompatibleAlertNotice = incompatibleAlertNoticeFn(() =>
+      this.setState({incompatibleAlertNotice: null})
```

Comment:
This is a bit of an unusual pattern, but I can see why it's needed. I think this is good to merge in, but we can chat a bit about another pattern that'd be useful here (using render props).